ci/scripts/r_build.sh

```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env bash
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+set -ex
+
+: ${R_BIN:=R}
+source_dir=${1}/r
+with_docs=${2:-false}
+
+pushd ${source_dir}
+
+${R_BIN} CMD INSTALL .
+
+if [ "${with_docs}" == "true" ]; then
```

Comment:
This feels hacky. Why not add a new script to build the docs instead of polluting this script?

Reply:
I know, your suggestion was my original idea. The issue is with the C GLib documentation [generation](https://github.com/apache/arrow/blob/master/ci/scripts/c_glib_build.sh#L36) which is handled with the same meson command in the same script. To remain consistent with that the doc building functionality is built in the *_build.sh scripts.

@kou could we factor out the glib doc generation to c_glib_docs.sh or something?

Reply:
docs_build.sh already exists and that's where C++ and Python docs are generated. I'd put this in that script, right before the `rsync`. Whether or not you move the other doc generating to `docs_build.sh` or whether those remain as just rsync seems like a separate issue (you could try to clean that up if possible but seems not required)

Reply:
`docs_build.sh` only builds the sphinx documentations (which is native to python) but for example doxygen is generated from the cpp_build.sh script.

Reply:
I'd rather keep this as is because of consistency, and we can figure out a better way to organize these scripts later.